eng/pipelines/templates/steps/cache-maven-repository.yml

```diff
@@ -1,6 +1,17 @@
+parameters:
+  - name: JobType
+    type: string
+    default: test
+  - name: SDKType
+    type: string
+    default: client
+  - name: ServiceDirectory
+    type: string
+    default: 'not-specified' # Set a default that breaks in obvious ways.
+
 steps:
 - task: Cache@2
   inputs:
-    key: 'maven | $(CacheSalt) | $(Agent.OS) | $(Build.SourcesDirectory)/eng/versioning/external_dependencies.txt'
+    key: 'maven | "$(CacheSalt)" | "$(Agent.OS)" | $(Build.SourcesDirectory)/eng/versioning/external_dependencies.txt | "${{ parameters.SDKType }}" | "${{ parameters.ServiceDirectory }}" | "${{ parameters.JobType }}" | "$(TestFromSource)"'
```

Comment:
Also the extra double quoting isn't necessary as the entire key turns into a single string.

Reply:
Quoting was done as some steps do not pass all parameters as non-null. Without the quotes it resulted in the pipeline having a null pointer exception being thrown, with quotes it ended up as `""`.